api/v1alpha1/httproute_types.go

```diff
@@ -43,6 +43,16 @@ type HTTPRouteHost struct {
 	// Rules are a list of HTTP matchers, filters and actions.
 	Rules []HTTPRouteRule `json:"rules"`
 
+	// TLS is the TLS configuration of an HTTPRouteHost.
+	//
+	// If unspecified, the TLS configuration of the binding gateway
+	// will be used.
```

Comment:
The gateway gets a server TLS config, but here we need a client TLS config because this is the TLS policy describing how to connect to a backend service, right? If so, how can we re-use the gateway policy?

Reply:
Kubernetes is adding support for describing app-protocol on the `core.Service`.
I would like to see if that suffices to describe the basic use case.

Reply:
https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20191227-app-protocol.md

Reply:
> we need a client TLS config because this is the TLS policy describing how to connect to a backend service, right?

That should be part of a HTTPRouteAction since you can have different TLS configs for different services.

@bowei The AppProtocol solves the problem of which protocol to use, but an operator can still want to specify protocol level configuration settings, such as CAs to use, client cert to use (if need be).

Having said that, my interpretation is that this field represents downstream TLS configs(TLS settings between client and proxy) and not upstream TLS config (between proxy and service).

Reply:
OK, I got mixed up here. I was thinking this was the TLS context for the upstream hop. but it's the server context for termination the client TLS session.

Reply:
One important thing to keep in mind is that while all the bells and whistles are really cool to have, for the initial version, we don't/can't/don't have to do it all.
Remember, there is a `core` that needs to be as universal as Ingress, `extended` that we can reasonably live with.
Things beyond that we should really try to aim towards understanding how they would be expressed as `custom`.

As things progress, it's easier to add than to remove.